runtime/src/main/java/org/corfudb/runtime/GarbageInformer.java

```diff
@@ -156,10 +160,28 @@ public void submitGCTask() {
         }
     }
 
+    @VisibleForTesting
+    public void waitUntilAllTasksFinish() {
+        // send garbage decisions to logUnit servers
+        while (getGarbageReceivingQueue().size() > 0) {
+            gcUnsafe();
+        }
+        
+        // waits until all task in the executor finish
+        drainExecutor.shutdown();
+        try {
+            drainExecutor.awaitTermination(TERMINATION_WAIT_TIME.getSeconds(), TimeUnit.SECONDS);
+        } catch (InterruptedException ex) {
+            log.error("Encounter interruption exception when {} is await termination", drainExecutor, ex);
+            throw new UnrecoverableCorfuInterruptedError(ex);
+        } finally {
+            drainExecutor.shutdownNow();
```

Comment:
You need to throw any exception happened in this method, I think add `throw Exception` in the method signature and remove shutdownNow would be better.